gdeep/trainer/trainer.py

```diff
@@ -304,7 +320,23 @@ def closure() -> Tensor:
             pred, X, y = self._send_to_device(X, y)
             batch_metric = self.training_metric(pred, y)
             metric_list.append(batch_metric)
-            loss = self.loss_fn(pred, y)
+            if self.regularize == True:
+                loss = self.loss_fn(pred, y)
+                self.regularizer.regularizer_arguments["model"] = copy.deepcopy(
+                    self.model
+                )
+                self.regularizer.regularizer_arguments[
```

Comment:
What's the type of self.regularizer. The default value is "Regularizer"`.

Reply:
I believe the type is `Regularizer`, as per line 39 and 148.

Reply:
correct